### DISCUSSION TITLE
Error Installing AdGuardHome on UDMP

### DISCUSSION THREAD

**Discussion by whitedragon551** [unifi-utilities/unifi-common#210] [Q&A]
Trying to install AdGuardHome on my UDMP using: https://github.com/boostchicken/udm-utilities/tree/master/AdguardHome

At step 2 and when I enter the command: /mnt/data/on_boot.d/05-install-cni-plugins.sh from a putty connection as root on the UDMP, I get error: 

# -sh: /mnt/data/on_boot.d/05-install-cni-plugins.sh: Permission denied

How do I fix this? I created the 05-install-cni-plugins.sh file by going to: https://github.com/boostchicken/udm-utilities/blob/master/cni-plugins/05-install-cni-plugins.sh and copying and pasting the info into notepad++ and saving with the name. Then moved it over to the UDMP file location with winscp.

**Comment by FlorentCoppint** [comment]
You don't know how to use linux ? Maybe learn permissions...
chmod u+x /mnt/data/on_boot.d/05-install-cni-plugins.sh

**Reply by whitedragon551** [reply]
FWIW I did do that and here are the results.

chmod u+x /mnt/data/on_boot.d/05-install-cni-plugins.sh
/mnt/data/on_boot.d/05-install-cni-plugins.sh
-sh: /mnt/data/on_boot.d/05-install-cni-plugins.sh: not found
cd /mnt/data/on_boot.d
ls
05-install-cni-plugins.sh  10-dns.sh
05-install-cni-plugins.sh
-sh: 05-install-cni-plugins.sh: not found

The file is clearly there with proper permissions.

**Comment by pietro-v** [comment]
./05-install-cni-plugins.sh 
when in the directory has the same result?

**Reply by whitedragon551** [reply]
Got it running last night. I'm seeing weird wireless speed negotiations now. Devices that would normally run on ac bands are now pushes down to n or slower.
